pkg/tests/schema_pulumi_test.go

```diff
@@ -4159,3 +4012,3298 @@ func TestMakeTerraformResultNilVsEmptyMap(t *testing.T) {
+			},
+			autogold.Expect(`Previewing update (test):
+  pulumi:pulumi:Stack: (same)
+    [urn=urn:pulumi:test::test::pulumi:pulumi:Stack::test-test]
+    ~ prov:index/test:Test: (update)
+        [id=newid]
+        [urn=urn:pulumi:test::test::prov:index/test:Test::mainRes]
+      ~ tests: [
+          + [1]: "val5"
+          ~ [2]: "val3" => "val6"
+          - [3]: "val4"
+        ]
+Resources:
+    ~ 1 to update
+    1 unchanged
+`),
+			map[string]interface{}{
+				"tests[1]": map[string]interface{}{"kind": "ADD_REPLACE"},
+				"tests[2]": map[string]interface{}{"kind": "UPDATE_REPLACE"},
+				"tests[3]": map[string]interface{}{"kind": "DELETE_REPLACE"},
+			},
+			autogold.Expect(`Previewing update (test):
+  pulumi:pulumi:Stack: (same)
+    [urn=urn:pulumi:test::test::pulumi:pulumi:Stack::test-test]
+    +-prov:index/test:Test: (replace)
+        [id=newid]
+        [urn=urn:pulumi:test::test::prov:index/test:Test::mainRes]
+      ~ tests: [
+          + [1]: "val5"
+          ~ [2]: "val3" => "val6"
+          - [3]: "val4"
+        ]
+Resources:
+    +-1 to replace
+    1 unchanged
+`),
+			map[string]interface{}{
+				"tests[1]":        map[string]interface{}{},
+				"tests[2].nested": map[string]interface{}{"kind": "UPDATE"},
+				"tests[3]":        map[string]interface{}{"kind": "DELETE"},
+			},
+			autogold.Expect(`Previewing update (test):
+  pulumi:pulumi:Stack: (same)
+    [urn=urn:pulumi:test::test::pulumi:pulumi:Stack::test-test]
+    ~ prov:index/test:Test: (update)
+        [id=newid]
+        [urn=urn:pulumi:test::test::prov:index/test:Test::mainRes]
+      ~ tests: [
+          + [1]: {
+                  + nested    : "val5"
+                }
+          ~ [2]: {
+                  ~ nested: "val3" => "val6"
+                }
+          - [3]: {
+                  - nested: "val4"
+                }
+        ]
+Resources:
+    ~ 1 to update
+    1 unchanged
+`),
+			map[string]interface{}{
+				"tests[1]":        map[string]interface{}{"kind": "ADD_REPLACE"},
+				"tests[2].nested": map[string]interface{}{"kind": "UPDATE_REPLACE"},
+				"tests[3]":        map[string]interface{}{"kind": "DELETE_REPLACE"},
+			},
+			autogold.Expect(`Previewing update (test):
+  pulumi:pulumi:Stack: (same)
+    [urn=urn:pulumi:test::test::pulumi:pulumi:Stack::test-test]
+    +-prov:index/test:Test: (replace)
+        [id=newid]
+        [urn=urn:pulumi:test::test::prov:index/test:Test::mainRes]
+      ~ tests: [
+          + [1]: {
+                  + nested    : "val5"
+                }
+          ~ [2]: {
+                  ~ nested: "val3" => "val6"
+                }
+          - [3]: {
+                  - nested: "val4"
+                }
+        ]
+Resources:
+    +-1 to replace
+    1 unchanged
+`),
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			for _, forceNew := range []bool{false, true} {
+				t.Run(fmt.Sprintf("ForceNew=%v", forceNew), func(t *testing.T) {
```

Comment:
Doing less in each one test case would be more readable maybe? One test for ForceNew, one for BLock, etc combination you are seeking to cover.

Reply:
That'd lose the connection between these tests - they are meant to be very similar. I've trimmed the recorded output to only the essential part of the tests. I think this makes the tests a lot more readable, LMK what you think.